back/lib/error_reporter.rb

```diff
@@ -4,17 +4,25 @@
 # https://github.com/rails/rails/pull/43625/files
 class ErrorReporter
   class << self
-    # Report any unhandled exception, and swallow it.
-    #
-    #  ErrorReporter.handle do
-    #    1 + '1'
-    #  end
-    #
-    def handle(error_class = StandardError)
-      yield
-    rescue error_class => e
-      report(e)
-      nil
+    # Do not swallow exceptions in test and development environments.
+    # Ideally, we should not rely on `Rails` constant since this file is in the lib folder.
+    if Rails.env.test? || Rails.env.development?
+      def handle(_error_class = nil)
+        yield
+      end
```

Comment:
I follow the logic, but it does risk having larger difference in behavior between the environments. In particular in the case where someone is reporting an error they want to track, but shouldn't necessary break.

Maybe we don't want this system to be used for such reasons, but still somewhat doubting whether 'enforcing' it like this might not lead to harder debugging, at some point.

Reply:
I see your point. I’m okay with adding it back for `Rails.env.development`, but I still think the test suite shouldn’t make requests to external services. IMO, we should stub or spy on the `handle` method instead.